core/src/services/cloudflare_kv/backend.rs

```diff
@@ -0,0 +1,444 @@
+            client
+        } else {
+            HttpClient::new().map_err(|err| {
+                err.with_operation("Builder::build")
+                    .with_context("service", Scheme::CloudflareKv)
+            })?
+        };
+
+        let root = normalize_root(
+            self.root
+                .clone()
+                .unwrap_or_else(|| "/".to_string())
+                .as_str(),
+        );
+
+        Ok(kv::Backend::new(Adapter {
+            authorization,
+            account_id,
+            namespace_id,
+            client,
+        })
+        .with_root(&root))
+    }
+}
+
+pub type CloudflareKvBackend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    authorization: Option<String>,
+    account_id: String,
+    namespace_id: String,
+    client: HttpClient,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Adapter")
+            .field("account_id", &self.account_id)
+            .field("namespace_id", &self.namespace_id)
+            .finish()
+    }
+}
+
+impl Adapter {
+    fn load_token<T>(&self, mut req: Request<T>) -> Result<Request<T>> {
+        if let Some(auth) = &self.authorization {
+            req.headers_mut()
+                .insert(header::AUTHORIZATION, auth.parse().unwrap());
+        }
+        Ok(req)
+    }
+
+    #[inline]
+    pub async fn send(&self, req: Request<AsyncBody>) -> Result<Response<IncomingAsyncBody>> {
+        self.client.send(req).await
+    }
+}
+
+#[async_trait]
+impl kv::Adapter for Adapter {
+    fn metadata(&self) -> kv::Metadata {
+        kv::Metadata::new(
+            Scheme::CloudflareKv,
+            &self.namespace_id,
+            Capability {
+                read: true,
+                write: true,
+                list: true,
+
+                ..Default::default()
+            },
+        )
+    }
+
+    async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
+        let url = format!(
+            r"https://api.cloudflare.com/client/v4/accounts/{}/storage/kv/namespaces/{}",
+            self.account_id, self.namespace_id
+        );
+        let url = format!("{}/values/{}", url, path);
+        let mut req = Request::get(&url);
+        req = req.header(header::CONTENT_TYPE, "application/json");
+        let mut req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        req = self.load_token(req)?;
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => {
+                let body = resp.into_body().bytes().await?;
+                Ok(Some(body.into()))
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn set(&self, path: &str, value: &[u8]) -> Result<()> {
+        let url = format!(
+            r"https://api.cloudflare.com/client/v4/accounts/{}/storage/kv/namespaces/{}",
```

Comment:
Maybe we can extract `https://api.cloudflare.com/client/v4` as a constant? Or can store the url `https://api.cloudflare.com/client/v4/accounts/{account_id}/storage/kv/namespaces/{namespace_id}` in adapter instead of build it everytime.